apps/explorer/src/app/components/blocks/blocks-infinite-list.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { FixedSizeList as List } from 'react-window';
+import InfiniteLoader from 'react-window-infinite-loader';
+import { t } from '@vegaprotocol/react-helpers';
+import type { BlockMeta } from '../../routes/blocks/tendermint-blockchain-response';
+import { BlockData } from './block-data';
+
+interface BlocksInfiniteListProps {
+  hasMoreBlocks: boolean;
+  areBlocksLoading: boolean | undefined;
+  blocks: BlockMeta[] | undefined;
+  loadMoreBlocks: () => void;
+  error: Error | undefined;
+  className?: string;
+}
+
+interface ItemProps {
+  index: number;
+  style: React.CSSProperties;
+}
+
+export const BlocksInfiniteList = ({
+  hasMoreBlocks,
+  areBlocksLoading,
+  blocks,
+  loadMoreBlocks,
+  error,
+  className,
+}: BlocksInfiniteListProps) => {
+  if (!blocks) {
+    return <div>No items</div>;
+  }
+
+  // If there are more items to be loaded then add an extra row to hold a loading indicator.
+  const itemCount = hasMoreBlocks ? blocks.length + 1 : blocks.length;
+
+  // Pass an empty callback to InfiniteLoader in case it asks us to load more than once.
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const loadMoreItems = areBlocksLoading ? () => {} : loadMoreBlocks;
```

Comment:
```suggestion
  const loadMoreItems = areBlocksLoading ? NOOP : loadMoreBlocks;
```

and at the top of the file in module scope `const NOOP = () => {}`